File: hosts/macos/Sources/EivizMac/InputResolver.swift
import Foundation

enum InputResolver {
    static func resolve(_ session: MixerSessionData, key: String) -> InputEntry? {
        let trimmed = key.trimmingCharacters(in: .whitespacesAndNewlines)
        guard !trimmed.isEmpty else { return nil }
        if let guid = UUID(uuidString: trimmed) {
            let text = guid.uuidString
            return session.inputs.first {
                $0.guid.caseInsensitiveCompare(text) == .orderedSame
                    || $0.guid.caseInsensitiveCompare(trimmed) == .orderedSame
            }
        }
        if let number = UInt64(trimmed) {
            if let byId = session.inputs.first(where: { $0.id == number }) {
                return byId
            }
            if number >= 1 && number <= UInt64(session.inputs.count) {
                return session.inputs[Int(number) - 1]
            }
        }
        return session.inputs.first { $0.name.caseInsensitiveCompare(trimmed) == .orderedSame }
    }
}

File: hosts/win32/InputResolver.cs
namespace Eiviz.Host;

internal static class InputResolver
{
    public static InputEntry? Resolve(Session session, string key)
    {
        if (string.IsNullOrWhiteSpace(key))
            return null;
        key = key.Trim();
        if (System.Guid.TryParse(key, out var guid))
        {
            var text = guid.ToString();
            return session.Inputs.FirstOrDefault(item =>
                string.Equals(item.Guid, text, StringComparison.OrdinalIgnoreCase)
                || string.Equals(item.Guid, key, StringComparison.OrdinalIgnoreCase));
        }
        if (ulong.TryParse(key, out var number))
        {
            var byId = session.Inputs.FirstOrDefault(item => item.Id == number);
            if (byId is not null)
                return byId;
            if (number >= 1 && number <= (ulong)session.Inputs.Count)
                return session.Inputs[(int)number - 1];
        }
        return session.Inputs.FirstOrDefault(item =>
            string.Equals(item.Name, key, StringComparison.OrdinalIgnoreCase));
    }
}
